src/server/engine_shard_set.h

```diff
@@ -329,14 +335,18 @@ void EngineShardSet::RunBriefInParallel(U&& func, P&& pred) const {
   bc.Wait();
 }
 
-template <typename U> void EngineShardSet::RunBlockingInParallel(U&& func) {
-  BlockingCounter bc{size()};
+template <typename U, typename P> void EngineShardSet::RunBlockingInParallel(U&& func, P&& pred) {
+  BlockingCounter bc{0};
   static_assert(std::is_invocable_v<U, EngineShard*>,
                 "Argument must be invocable EngineShard* as argument.");
   static_assert(std::is_void_v<std::invoke_result_t<U, EngineShard*>>,
                 "Callable must not have a return value!");
 
   for (uint32_t i = 0; i < size(); ++i) {
+    if (!pred(i))
```

Comment:
This a predicate skip mechanism correct? (if pred is false, we skip dispatching the function fun and consequently we don't block/wait for it at the end -- since it never gets dispatched)

Reply:
Yes. This is exactly like `RunBriefInParallel` above (actually we could unite them into one function with one more template param, but I know not everybody likes excessive template usage 🙂 )